app/javascript/packs/qr_code_scan.js

```diff
@@ -0,0 +1,132 @@
+import QrScanner from 'qr-scanner';
+import QrScannerWorkerPath from '!!file-loader!../../..//node_modules/qr-scanner/qr-scanner-worker.min.js';
+QrScanner.WORKER_PATH = QrScannerWorkerPath;
+
+const healthCardPattern = /^shc:\/(?<multipleChunks>(?<chunkIndex>[0-9]+)\/(?<chunkCount>[0-9]+)\/)?[0-9]+$/;
+
+let qrScanner;
+let startButton;
+let stopButton;
+let successNotification;
+let errorNotification;
+let inputField;
+let multiStatusContainer;
+
+let scannedCodes = [];
+
+document.addEventListener('DOMContentLoaded', () => {
+  const videoElement = document.getElementById('preview');
+
+  qrScanner = new QrScanner(videoElement, handleScan);
+
+  startButton = document.getElementById('start');
+  stopButton = document.getElementById('stop');
+  successNotification = document.getElementById('success-notification');
+  errorNotification = document.getElementById('error-notification');
+  inputField = document.getElementById('qr-contents');
+  multiStatusContainer = document.getElementById('multi-status-container');
+
+  startButton.addEventListener('click', startScanning);
+  stopButton.addEventListener('click', stopScanning);
+});
+
+const startScanning = () => {
+  disableStartButton();
+  hideSuccessNotification();
+  hideErrorNotification();
+  qrScanner.start();
+};
+
+const stopScanning = () => {
+  enableStartButton();
+  qrScanner.stop();
+};
+
+const disableStartButton = () => {
+  startButton.setAttribute('disabled', true);
+  stopButton.removeAttribute('disabled');
+};
+
+const enableStartButton = () => {
+  stopButton.setAttribute('disabled', true);
+  startButton.removeAttribute('disabled');
+};
+
+const hideSuccessNotification = () => {
+  successNotification.setAttribute('hidden', true);
+};
+
+const showSuccessNotification = () => {
+  hideErrorNotification();
+  successNotification.removeAttribute('hidden');
+};
+
+const hideErrorNotification = () => {
+  errorNotification.setAttribute('hidden', true);
+};
+
+const showErrorNotification = () => {
+  hideSuccessNotification();
+  errorNotification.removeAttribute('hidden');
+};
+
+const handleScan = result => {
+  console.log(result);
+
+  if(healthCardPattern.test(result)) {
+    const match = result.match(healthCardPattern);
+    if(match.groups.multipleChunks) {
+      hideErrorNotification();
+      const chunkCount = +match.groups.chunkCount;
+      const currentChunkIndex = +match.groups.chunkIndex;
+      if(scannedCodes.length !== chunkCount) {
+        scannedCodes = new Array(chunkCount);
+        for(let i = 0; i < chunkCount; i++) {
+          scannedCodes[i] = null;
+        }
```

Comment:
Why does this Array need to be populated with `null`, instead of the leaving the entries as `undefined` (which is also falsy)?

If this is needed, could we use [`fill()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/fill) instead?

Reply:
Yeah, we can use `fill`. Even though the array is created with a specific length, and returns that length if you ask for it, it won't iterate over those elements without them being initialized.

![Screen Shot 2021-04-01 at 2 40 26 PM](https://user-images.githubusercontent.com/15969967/113339254-398cb700-92f8-11eb-829b-8ac228ceb809.png)